### PR TITLE
fix: prevent node agent control-plane scheduling

### PR DESCRIPTION
**Pull request by Levi080513**
## Issue

NEU-549

## Summary

Prevent `neutree-node-agent` from scheduling on Kubernetes control-plane nodes tainted with `node-role.kubernetes.io/control-plane:NoSchedule`.

## Changes

- Remove the unconditional `operator: Exists` toleration from the controller-rendered `neutree-node-agent` DaemonSet.
- Add a rendered-resource regression test that requires an empty node-agent tolerations list.
- Add conditional Kubernetes E2E coverage for control-plane exclusion and untainted worker placement (TestRail `C2739216`).
- Limit the change to the community implementation; enterprise source, sync, build, and runtime validation are out of scope.

### Unit test

- [x] `go test ./internal/cluster/component/metrics -count=1`
- [x] `go vet ./internal/cluster/component/metrics ./tests/e2e`

### DB test

- Not affected: no schema, query, or persistence behavior changed.

### E2E test

- [x] Manual TestRail `C2739216`: on a six-node leased Kubernetes environment, node-agent rendered with no tolerations, targeted the three untainted workers, and had no Pods on the three tainted control-plane nodes.

## Risk & Rollback

Custom tainted worker nodes no longer receive node-agent automatically; this is the approved behavior for NEU-549. Reverting this PR restores the previous broad toleration behavior.

### CODE REVIEW

File: tests/e2e/cluster_metrics_helper_test.go [hunk 1]
```diff
@@ -127,6 +128,106 @@ func assertK8sMetricsResources(
...
+	health, err := k8sH.PodProxyGetRaw(ctx, namespace, readyWorkerPod.Name, "19101", "/health")
+	ExpectWithOffset(1, err).NotTo(HaveOccurred(), "node-agent /health should be reachable on an untainted worker")
+	ExpectWithOffset(1, string(health)).To(ContainSubstring(`"status":"ok"`))
+}
+
+func hasControlPlaneNoScheduleTaint(node corev1.Node) bool {
+	for _, taint := range node.Spec.Taints {
+		if taint.Key == "node-role.kubernetes.io/control-plane" && taint.Effect == corev1.TaintEffectNoSchedule {
+			return true
+		}
```
[the remark this answers is not in the document]
Comment (Levi080513) [reply tests/e2e/cluster_metrics_helper_test.go]: Fixed in 6926ecd9: legacy node-role.kubernetes.io/master:NoSchedule is now treated as a control-plane target, with table-driven coverage for both supported keys and non-matching taints.

**Review by Yuyz0112 — APPROVED** [review]
